hist/hist/src/TF1.cxx

```diff
@@ -2073,11 +2073,100 @@ Int_t TF1::GetQuantiles(Int_t nprobSum, Double_t *q, const Double_t *probSum)
+   Int_t intNegative = 0;
+   Int_t i;
+   Bool_t logbin = kFALSE;
+   Double_t dx;
+   Double_t xmin = fXmin;
+   Double_t xmax = fXmax;
+   TString opt(option);
+   opt.ToUpper();
+   // perform a log binning if specified by user (option="Log") or if some conditions are met
+   // and the user explicitly does not specify a Linear binning option
+   if (opt.Contains("LOG") || ((xmin > 0 && xmax / xmin > fNpx) && !opt.Contains("LIN"))) {
+      logbin = kTRUE;
+      fAlpha[fNpx] = 1;
+      xmin = TMath::Log10(fXmin);
+      xmax = TMath::Log10(fXmax);
+      if (gDebug)
+         Info("GetRandom", "Use log scale for tabulating the integral in [%f,%f] with %d points", fXmin, fXmax, fNpx);
+   }
+   dx = (xmax - xmin) / fNpx;
+
+   std::vector<Double_t> xx(fNpx + 1);
+   for (i = 0; i < fNpx; i++) {
+      xx[i] = xmin + i * dx;
+   }
+   xx[fNpx] = xmax;
+   for (i = 0; i < fNpx; i++) {
+      if (logbin) {
+         integ = Integral(TMath::Power(10, xx[i]), TMath::Power(10, xx[i + 1]), 0.0);
+      } else {
+         integ = Integral(xx[i], xx[i + 1], 0.0);
+      }
+      if (integ < 0) {
+         intNegative++;
+         integ = -integ;
+      }
+      fIntegral[i + 1] = fIntegral[i] + integ;
+   }
+   if (intNegative > 0) {
+      Warning("GetRandom", "function:%s has %d negative values: abs assumed", GetName(), intNegative);
+   }
+   if (fIntegral[fNpx] == 0) {
+      Error("GetRandom", "Integral of function is zero");
+      return kFALSE;
+   }
+   Double_t total = fIntegral[fNpx];
+   for (i = 1; i <= fNpx; i++) { // normalize integral to 1
+      fIntegral[i] /= total;
+   }
+   // the integral r for each bin is approximated by a parabola
+   //  x = alpha + beta*r +gamma*r**2
+   // compute the coefficients alpha, beta, gamma for each bin
+   Double_t x0, r1, r2, r3;
+   for (i = 0; i < fNpx; i++) {
+      x0 = xx[i];
+      r2 = fIntegral[i + 1] - fIntegral[i];
+      if (logbin)
+         r1 = Integral(TMath::Power(10, x0), TMath::Power(10, x0 + 0.5 * dx), 0.0) / total;
+      else
+         r1 = Integral(x0, x0 + 0.5 * dx, 0.0) / total;
+      r3 = 2 * r2 - 4 * r1;
+      if (TMath::Abs(r3) > 1e-8)
+         fGamma[i] = r3 / (dx * dx);
+      else
+         fGamma[i] = 0;
+      fBeta[i] = r2 / dx - fGamma[i] * dx;
+      fAlpha[i] = x0;
+      fGamma[i] *= 2;
+   }
+   return kTRUE;
+}
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Return a random number following this function shape
 ///
+/// @param rng  Ranodm numjber generator. By default (or when passing a nullptr) the global gRandom is used
+/// @param option Option string which controls the binning used to compute the integral. Default mode is automatic depending of
+///               xmax, xmin and Npx (function points).
+///               Possible values are: "LOG" to force usage of log scale for tabulating the integral
+///                                    "LIN" to force usage of linear scale when tabulating the integral
```

Comment:
```suggestion
/// @param rng  Random number generator. By default (or when passing a nullptr), the global `gRandom` is used
/// @param option Option string which controls the binning used to compute the integral. Default mode is automatic depending of
///               xmax, xmin and Npx (function points).
///               Possible values are:
///               - "LOG" to force usage of log scale for tabulating the integral
///               - "LIN" to force usage of linear scale when tabulating the integral
```
Note that doxygen will collapse all indentation, so bullet points are necessary.